gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java

```diff
@@ -0,0 +1,296 @@
+		}
+
+		if (!project.isPresent()) {
+			throw new IllegalArgumentException("Project '" + projectName + "' was not found");
+		}
+
+		return project.get().getHttpUrl();
+	}
+
+	private GitProject createAndInitGitLabProject() {
+		GitProject project;
+		if (GitLabConfig.Protocol.HTTP.equals(GitLabConfig.protocol())) {
+			project = GitUtil.initRepository(gitRepoLocalPath, gitRepoHttpUrl,
+					GitLabConfig.username(),
+					GitLabConfig.password());
+		} else {
+			project = GitUtil.initRepository(gitRepoLocalPath, gitRepoSshUrl);
+		}
+		project.setHttpUrl(gitRepoHttpUrl);
+		return project;
+	}
+
+	public GitProject createProject() {
+		return createProject(null);
+	}
+
+	public GitProject createProject(String name) {
+		initGitLabProject(name, null);
+
+		return createAndInitGitLabProject();
+	}
+
+	public GitProject createProjectFromPath(String name, Path path) {
+		return createProjectFromPath(name, path, null);
+	}
+
+	public GitProject createProjectFromPath(Path path, GitProjectModifier gitProjectModifier) {
+		return createProjectFromPath(null, path, gitProjectModifier);
+	}
+
+	public GitProject createProjectFromPath(String name, Path path, GitProjectModifier gitProjectModifier) {
+		return createProjectFromPath(name, path, gitProjectModifier, null);
+	}
+	
+	public GitProject createProjectFromPath(String name, Path path,	GitProjectModifier gitProjectModifier, Consumer<GitProject> postInitModifier) {
+		initGitLabProject(name, null);
+
+		Path sourcePath = path.toAbsolutePath();
+
+		try {
+			FileUtils.copyDirectory(sourcePath.toFile(), gitRepoLocalPath.toFile());
+		} catch (IOException e) {
+			LOGGER.error("Error when copying files to the local repository", e);
+			return null;
+		}
+
+		if(gitProjectModifier != null) {
+			gitProjectModifier.modify(sourcePath, gitRepoLocalPath);
+		}
+
+		GitProject project = createAndInitGitLabProject();
+
+		if (postInitModifier != null) {
+			postInitModifier.accept(project);
+		}
+		
+		project.addAll();
+		project.commit("initial commit");
+		project.push();
+
+		LOGGER.debug("Project \"" + name + "\" was created");
+		return project;
+	}
+
+	private void initGitLabProject(String name, String url) {
+		if (name == null || name.length() == 0) {
+			name = UUID.randomUUID().toString();
+		} else {
+			name = name + "-" + Integer.toString(Math.abs(new Random().nextInt()), 36);
+		}
+
+		try {
+			GitlabProject project = null;
+			int attempt = 0;
+			IOException lastExcepion = null;
+			while(project == null && attempt < WAIT_FOR_GITLAB_CREATE_PROJECT_MAX_ATTEMPTS) {
+				++attempt;
+				try {
+					Integer groupId = getGroupId();
+					project = api.createProject(name, groupId, null, null, null, null, null, null, true,
+							null, url);
+				} catch (IOException x) {
+					lastExcepion = x;
+					if (!x.getMessage().contains(
+							WAIT_FOR_GITLAB_CREATE_PROJECT_MESSAGE)
+							|| attempt >= WAIT_FOR_GITLAB_CREATE_PROJECT_MAX_ATTEMPTS) {
+						throw x;
+					} else {
+						LOGGER.debug(
```

Comment:
Could we make this one liner using standard logger syntax? log.debug("Some code with env value: {}", envValue);